include/ginkgo/core/base/batch_lin_op.hpp

```diff
@@ -697,7 +694,8 @@ class EnableBatchScaling : public BatchScalable {
          * BatchLinOp so that the following dynamic_cast is not needed.
```

Comment:
Can we have a `constexpr` type static cast here, which tries to check if the type is a `BatchLinOp` at compile time, so that this dynamic_cast is not needed anymore ?

Reply:
I don't think I know about this. Is that even possible? I thought the only way would be to make this a CRTP class, but that seems a bit overkill.